src/frontend/src/lib/stores/storage.store.ts

```diff
@@ -1,24 +1,27 @@
 import { del, get, set as setStorage } from '$icp/utils/storage.utils';
 import type { Option } from '$lib/types/utils';
-import { writable, type Readable } from 'svelte/store';
+import { writable, type Readable, type Writable } from 'svelte/store';
 
 export type StorageStoreData<T> = Option<T>;
 
-export interface StorageStore<T> extends Readable<StorageStoreData<T>> {
+export interface StorageStore<T>
+	extends Readable<StorageStoreData<T>>,
+		Pick<Writable<StorageStoreData<T>>, 'update'> {
 	set: (params: { key: string; value: T }) => void;
 	reset: (params: { key: string }) => void;
 }
 
 export const initStorageStore = <T>({ key }: { key: string }): StorageStore<T> => {
 	const initialValue = get<T>({ key });
 
-	const { set, subscribe } = writable<StorageStoreData<T>>(initialValue);
+	const { set, subscribe, update } = writable<StorageStoreData<T>>(initialValue);
 
 	return {
 		set: ({ key, value }) => {
 			setStorage({ key, value });
 			set(value);
 		},
+		update,
```

Comment:
I think we should not leak "update" because a consumer can use it and the value does not get saved in the local storage.

Is there another way?

Reply:
i tried without it and it is really not what i would do... basically it uses `subscribe` and not in a functional way... let me think a bit and come back on this